src/main/java/io/xpring/ilp/DefaultIlpClient.java

```diff
@@ -0,0 +1,139 @@
+package io.xpring.ilp;
+
+import org.interledger.spsp.server.grpc.AccountServiceGrpc;
+import org.interledger.spsp.server.grpc.BalanceServiceGrpc;
+import org.interledger.spsp.server.grpc.CreateAccountRequest;
+import org.interledger.spsp.server.grpc.CreateAccountResponse;
+import org.interledger.spsp.server.grpc.GetAccountRequest;
+import org.interledger.spsp.server.grpc.GetAccountResponse;
+import org.interledger.spsp.server.grpc.GetBalanceRequest;
+import org.interledger.spsp.server.grpc.GetBalanceResponse;
+import org.interledger.spsp.server.grpc.IlpOverHttpServiceGrpc;
+import org.interledger.spsp.server.grpc.SendPaymentRequest;
+import org.interledger.spsp.server.grpc.SendPaymentResponse;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.primitives.UnsignedLong;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.StatusRuntimeException;
+import io.xpring.ilp.grpc.IlpJwtCallCredentials;
+import io.xpring.xrpl.XpringException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * A client that can create accounts, get accounts, get balances, and send ILP payments on a connector.
+ */
+public class DefaultIlpClient implements IlpClientDecorator {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultIlpClient.class);
+
+    private final BalanceServiceGrpc.BalanceServiceBlockingStub balanceServiceStub;
+    private final AccountServiceGrpc.AccountServiceBlockingStub accountServiceStub;
+    private final IlpOverHttpServiceGrpc.IlpOverHttpServiceBlockingStub ilpOverHttpServiceStub;
+
+    /**
+     * Initialize a new client with a configured URL
+     * @param grpcUrl : The gRPC URL exposed by Hermes
+     */
+    protected DefaultIlpClient(String grpcUrl) {
+        this(ManagedChannelBuilder
+          .forTarget(grpcUrl)
+          .usePlaintext()
+          .build()
+        );
+    }
+
+    /**
+     * Required-args Constructor, currently for testing.
+     *
+     * @param channel A {@link ManagedChannel}.
+     */
+    public DefaultIlpClient(final ManagedChannel channel) {
```

Comment:
I think you should shut down these channels on process termination to avoid zombie threads:
https://github.com/xpring-eng/xpring4j/blob/master/src/main/java/io/xpring/xrpl/DefaultXpringClient.java#L66